spec/jobs/dttp/sync_schools_job_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Dttp::SyncSchoolsJob, type: :job do
+  include ActiveJob::TestHelper
+
+  before do
+    allow(Dttp::RetrieveSchools).to receive(:call).with(request_uri: request_uri).and_return(school_list)
+  end
+
+  let(:school_one_hash) { ApiStubs::Dttp::School.attributes }
+  let(:school_two_hash) { ApiStubs::Dttp::School.attributes }
+
+  let(:request_uri) { nil }
+  let(:school_list) do
+    {
+      items: [school_two_hash, school_one_hash],
+      meta: { next_page_url: "https://example.com" },
+    }
+  end
+
+  subject { described_class.perform_now }
+
+  it "enqueues job with the next_page_url" do
+    expect {
+      subject
+    }.to have_enqueued_job(described_class).with("https://example.com")
+  end
+
+  it "creates a Dttp::School record for each unique school" do
+    expect {
+      subject
+    }.to change(Dttp::School, :count).by(2)
+  end
+
+  context "when Dttp::School exist" do
+    let!(:dttp_school_one) { create(:dttp_school, dttp_id: school_one_hash["accountid"]) }
+
+    it "adds new records" do
+      expect {
+        subject
+      }.to change(Dttp::School, :count).by(1)
+    end
+
+    it "updates the existing record" do
+      subject
```

Comment:
This example can't prove that the update happend since the school name is always the same given it's hard coded in `ApiStubs::Dttp::School.attributes`.

Reply:
This expectation is made against an initial factory object which is updated with the values from the stub, however, I've made a change to make that explicit https://github.com/DFE-Digital/register-trainee-teachers/commit/9b85a58a48d86bc12425663f3ada3584b1f1f8c1